lib/setup/setupUpgrade.js

```diff
@@ -64,57 +64,41 @@ function Upgrade(options) {
         if (!dependencies && !globalDependencies) {
             return '';
         }
-        let adapters;
-        if (dependencies) {
-            // if like [{"js-controller": ">=0.10.1"}]
-            if (dependencies instanceof Array) {
-                adapters = {};
-                dependencies.filter(a => typeof dependencies[a] !== 'string')
-                    .forEach(rule => Object.keys(rule).filter(adapter => !adapters[adapter]).forEach(adapter => adapters[adapter] = rule[adapter]));
-            } // if like {"js-controller": ">=0.10.1", "admin": ">=0.5.6"}
-            else {
-                adapters = dependencies;
-            }
-        }
 
-        if (globalDependencies) {
-            // if like [{"js-controller": ">=0.10.1"}]
-            if (globalDependencies instanceof Array) {
-                adapters = adapters || {};
-                globalDependencies.filter(a => typeof globalDependencies[a] !== 'string')
-                    .forEach(rule => Object.keys(rule).filter(adapter => !adapters[adapter]).forEach(adapter => adapters[adapter] = rule[adapter]));
-            } // if like {"js-controller": ">=0.10.1", "admin": ">=0.5.6"}
-            else {
-                adapters = adapters || {};
-                Object.assign(adapters, globalDependencies);
-            }
-        }
+        dependencies = tools.parseDependencies(dependencies);
+        globalDependencies = tools.parseDependencies(globalDependencies);
+        // currently just combine them
+        Object.assign(dependencies, globalDependencies);
```

Comment:
is that really making sense to combine them here ... normally wrong...hm

Reply:
I would do the same as with install ... so check objects for global

Reply:
but then it becomes async ...gggrrr

Reply:
but ok ...possible to manage ... do getObjectView as first in the methods that callcheckDepedencies and hand it over inside te method

Reply:
We need to do globalCheck für globalDeps (getObjectView) and local check (read io-pack) on local deps check, but not part of this PR

Reply:
kk